### PR TITLE
Handle I/O errors when writing in Emit and EmitDifference

### PR DESCRIPTION
**Pull request by jcouv**
This is a follow-up on https://github.com/dotnet/roslyn/pull/19360
Some calls to WriteContentTo were not yet guarded.
Fixes https://github.com/dotnet/roslyn/issues/20091

@AlekseyTs @tmat @dotnet/roslyn-compiler for review (15.6). Thanks

### CODE REVIEW

**Comment by tmat** [comment]
:shipit:

File: src/Compilers/Core/Portable/PEWriter/PeWriter.cs [hunk 1]
```diff
@@ -160,7 +160,14 @@ internal static class PeWriter
...
                     if (portablePdbStream != null)
                     {
-                        portablePdbBlob.WriteContentTo(portablePdbStream);
+                        try
+                        {
+                            portablePdbBlob.WriteContentTo(portablePdbStream);
+                        }
+                        catch (Exception e) when (!(e is OperationCanceledException))
+                        {
+                            throw new PeWritingException(e);
```
Comment (AlekseyTs) [inline src/Compilers/Core/Portable/PEWriter/PeWriter.cs]: >PeWritingException [](start = 38, length = 18)

Should this be a PdbWritingException?

**Comment by jcouv** [comment]
@VSadov @agocke for another review. Thanks

**Comment by agocke** [comment]
retest windows_release_vs-integration_prtest please

**Comment by jcouv** [comment]
retest windows_release_vs-integration_prtest please

**Comment by AlekseyTs** [comment]
LGTM

**Pull request merged**